engine/src/main/java/io/zeebe/engine/processing/streamprocessor/CommandProcessorImpl.java

```diff
@@ -44,17 +68,17 @@ public void processRecord(
       final TypedStreamWriter streamWriter) {
 
     entityKey = command.getKey();
-    final boolean shouldRespond = wrappedProcessor.onCommand(command, this, streamWriter);
+    final boolean shouldRespond = wrappedProcessor.onCommand(command, this);
```

Comment:
So the one thing that still makes me uncomfortable is this method.

I think what I would like to see here is something like:
`Either<RejectionTuple, List<FollowUpRecordTuple> wrappedProcessor.onCommand(command, this)`

Where `FollowUpRecordTuple` can be either a follow up event or a follow up command.

I feel this interface right now is too limiting, because essentially as a command conrol implementation I can only send one follow up event (and if I want to do something else I need to find a different way to get hold of state writers, command writers, etc).

At the same time, I recognize that changing this would expand the scope of this PR and is maybe unwise.

Reply:
As discussed, I agree but let's do this later